### PR TITLE
backport: bitcoin#21679, #21913 - RPC default argument

### PR DESCRIPTION
**Pull request by knst**
## What was done?
Backport bitcoin#21679, #21913 which keep default argument value in correct type and some extra fixes and apply fixes for Dash specific code.

## How Has This Been Tested?
Run unit/functional tests.

## Breaking Changes
It doesn't actually change behavior, just improve implementation of default arguments and help for RPC.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1932,14 +1932,8 @@ static RPCHelpMan getchaintips()
...
-    int nBranchMin = -1;
-    int nCountMax = INT_MAX;
-
-    if(!request.params[0].isNull())
-        nCountMax = request.params[0].get_int();
-
-    if(!request.params[1].isNull())
-        nBranchMin = request.params[1].get_int();
+    const int nBranchMin{request.params[0].isNull() ? -1 : request.params[0].get_int()};
+    int nCountMax{request.params[1].isNull() ? INT_MAX : request.params[1].get_int()};
```
Comment (UdjinM6) [inline src/rpc/blockchain.cpp:1936]: params: 0 - count, 1 - branchlen
```suggestion
    int nCountMax{request.params[0].isNull() ? INT_MAX : request.params[0].get_int()};
    const int nBranchMin{request.params[1].isNull() ? -1 : request.params[1].get_int()};
```

Comment (knst) [reply src/rpc/blockchain.cpp]: I force-pushed already a fix, please update a page

**Review by UdjinM6 — APPROVED** [review]
utACK 0c3c7326bb568b59e65503856e335d595303630d

**Review by PastaPastaPasta — APPROVED** [review]
utACK 0c3c7326bb568b59e65503856e335d595303630d

**Pull request merged**